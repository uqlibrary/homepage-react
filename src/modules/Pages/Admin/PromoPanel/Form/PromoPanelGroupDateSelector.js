/* eslint-disable no-unused-vars */
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { DateTimePicker } from '@material-ui/pickers';
import { default as locale } from 'locale/promopanel.locale';

const moment = require('moment');

const useStyles = makeStyles(theme => ({
    contentBox: {
        minWidth: '90%',
        paddingTop: 20,
        '& img': {
            maxWidth: 800,
            height: 800,
            border: '1px solid grey',
            textAlign: 'center',
        },
        '& li': {
            marginBottom: 10,
            padding: 10,
            '&:hover': {
                backgroundColor: theme.palette.secondary.main,
                transition: 'background-color 1s ease',
            },
            '& p': {
                marginBottom: 0,
                marginTop: 1,
            },
        },
        '& [aria-labelledby="lightboxTitle"]': {
            color: 'blue',
        },
    },
    dialogPaper: {
        // make the block take up more of the page
        width: 500,
    },
    link: {
        marginBottom: 10,
        marginRight: 10,
        cursor: 'pointer',
    },
}));

export const handleChange = (event, startDate, endDate, setSaveEnabled, setStartDate, setEndDate) => value => {
    if (event === 'start') {
        if (moment(value.format('YYYY-MM-DD HH:mm')).isAfter(moment(endDate))) {
            setSaveEnabled(false);
        } else {
            setSaveEnabled(true);
        }
        setStartDate(value.format('YYYY-MM-DD HH:mm'));
    } else {
        if (moment(startDate).isAfter(value.format('YYYY-MM-DD HH:mm'))) {
            setSaveEnabled(false);
        } else {
            setSaveEnabled(true);
        }
        setEndDate(value.format('YYYY-MM-DD HH:mm'));
    }
};

export const PromoPanelGroupDateSelector = ({
    isEditingDate,
    defaultStartDate,
    defaultEndDate,
    handleCloseGroupDate,
    handleSaveGroupDate,
    scheduleChangeIndex,
    scheduleGroupIndex,
    fullPromoPanelUserTypeList,
    setConfirmationMessage,
    setIsConfirmOpen,
    setConfirmationMode,
    panelScheduleId,
    displayList,
}) => {
    const classes = useStyles();

    const [startDate, setStartDate] = useState(defaultStartDate);
    const [endDate, setEndDate] = useState(defaultEndDate);
    const [saveEnabled, setSaveEnabled] = useState(true);

    useEffect(() => {
        setStartDate(defaultStartDate);
        setEndDate(defaultEndDate);
    }, [defaultStartDate, defaultEndDate]);

    // const handleChange = event => value => {
    //     if (event === 'start') {
    //         if (moment(value.format('YYYY-MM-DD HH:mm')).isAfter(moment(endDate))) {
    //             setSaveEnabled(false);
    //         } else {
    //             setSaveEnabled(true);
    //         }
    //         setStartDate(value.format('YYYY-MM-DD HH:mm'));
    //     } else {
    //         if (moment(startDate).isAfter(value.format('YYYY-MM-DD HH:mm'))) {
    //             setSaveEnabled(false);
    //         } else {
    //             setSaveEnabled(true);
    //         }
    //         setEndDate(value.format('YYYY-MM-DD HH:mm'));
    //     }
    // };

    const handleGroupDateClose = () => {
        setStartDate(defaultStartDate);
        setStartDate(defaultEndDate);
        handleCloseGroupDate();
    };

    const handleGroupDateSave = () => {
        let isValid = true;
        // Check against existing schedules already saved
        fullPromoPanelUserTypeList.map(schedules => {
            /* istanbul ignore else */
            if (schedules.usergroup_group === scheduleGroupIndex) {
                schedules.scheduled_panels &&
                    schedules.scheduled_panels.map(schedule => {
                        /* istanbul ignore else */
                        if (isValid && schedule.panel_schedule_id !== panelScheduleId) {
                            /* istanbul ignore else */
                            if (
                                (moment(startDate).isSameOrAfter(moment(schedule.panel_schedule_start_time)) &&
                                    moment(startDate).isBefore(moment(schedule.panel_schedule_end_time))) ||
                                (moment(schedule.panel_schedule_start_time).isSameOrAfter(moment(startDate)) &&
                                    /* istanbul ignore next */
                                    moment(schedule.panel_schedule_start_time).isBefore(moment(endDate)))
                            ) {
                                isValid = false;
                                setConfirmationMessage(
                                    locale.form.scheduleConflict.alert(
                                        scheduleGroupIndex,
                                        schedule.panel_title,
                                        schedule.panel_schedule_start_time,
                                        schedule.panel_schedule_end_time,
                                    ),
                                );
                            }
                        }
                    });
            }
        });
        displayList.map(alloc => {
            /* istanbul ignore else  */
            if (
                (moment(startDate).isSameOrAfter(moment(alloc.startDate)) &&
                    moment(startDate).isBefore(moment(alloc.endDate))) ||
                (moment(alloc.startDate).isSameOrAfter(moment(startDate)) &&
                    moment(alloc.startDate).isBefore(moment(endDate)) &&
                    isValid)
            ) {
                isValid = false;
                setConfirmationMessage(
                    locale.form.scheduleConflict.alert(
                        scheduleGroupIndex,
                        `Schedule existing in this panel for ${scheduleGroupIndex}`,
                        alloc.startDate,
                        alloc.endDate,
                    ),
                );
            }
        });

        if (isValid) {
            handleSaveGroupDate(scheduleChangeIndex, { start: startDate, end: endDate });
        } else {
            setConfirmationMode('schedule');
            setIsConfirmOpen(true);
        }
    };

    return (
        <React.Fragment>
            <Dialog
                open={isEditingDate}
                aria-labelledby="lightboxTitle"
                PaperProps={{ classes: { root: classes.dialogPaper } }}
            >
                <DialogTitle
                    id="lightboxTitle"
                    data-testid="panel-edit-date-title"
                    style={{ position: 'relative', borderBottom: '1px solid #d7d1cc', fontSize: 12 }}
                    children={<p style={{ lineHeight: 1, margin: 0 }}>{'Edit Schedule Dates'}</p>}
                />
                <DialogContent>
                    <Grid container spacing={1}>
                        <Grid item xs>
                            <DateTimePicker
                                id="admin-promopanel-group-start-date"
                                data-testid="admin-promopanel-group-start-date"
                                value={startDate}
                                label="Start date"
                                onChange={handleChange(
                                    'start',
                                    startDate,
                                    endDate,
                                    setSaveEnabled,
                                    setStartDate,
                                    setEndDate,
                                )}
                                format="DD/MM/YYYY HH:mm a"
                                showTodayButton
                                todayLabel={'Today'}
                                autoOk
                                KeyboardButtonProps={{
                                    'aria-label': 'Start Date',
                                    'data-testid': 'end-date-calendar',
                                }}
                            />
                            {moment(startDate).isBefore(moment().subtract(1, 'minutes')) && (
                                <div
                                    className={classes.errorStyle}
                                    id="start-date-warning"
                                    data-testid="start-date-warning"
                                >
                                    This date is in the past.
                                </div>
                            )}
                        </Grid>
                        <Grid item xs align="left">
                            <DateTimePicker
                                id="admin-promopanel-group-end-date"
                                data-testid="admin-promopanel-group-end-date"
                                value={endDate}
                                label="End date"
                                onChange={handleChange(
                                    'end',
                                    startDate,
                                    endDate,
                                    setSaveEnabled,
                                    setStartDate,
                                    setEndDate,
                                )}
                                format="DD/MM/YYYY HH:mm a"
                                showTodayButton
                                todayLabel={'Today'}
                                autoOk
                                KeyboardButtonProps={{
                                    'aria-label': 'Start Date',
                                    'data-testid': 'end-date-calendar',
                                }}
                            />
                            {moment(endDate).isBefore(moment().subtract(1, 'minutes')) && (
                                <div
                                    className={classes.errorStyle}
                                    id="end-date-warning"
                                    data-testid="end-date-warning"
                                >
                                    This date is in the past.
                                </div>
                            )}
                        </Grid>
                        <Grid item xs={12}>
                            {moment(endDate).isBefore(moment(startDate)) && (
                                <span data-testid="panel-start-date-warning">Start Date cannot be after End Date.</span>
                            )}
                        </Grid>

                        <Grid item xs={12} align="right">
                            <Button
                                style={{ marginTop: 10 }}
                                color="secondary"
                                children="Cancel"
                                data-testid="admin-promopanel-group-button-cancel"
                                variant="contained"
                                onClick={handleGroupDateClose}
                            />

                            <Button
                                style={{ marginTop: 10 }}
                                color="primary"
                                children="Ok"
                                disabled={!saveEnabled}
                                data-testid="admin-promopanel-group-button-save"
                                variant="contained"
                                onClick={handleGroupDateSave}
                            />
                        </Grid>
                    </Grid>
                </DialogContent>
            </Dialog>
        </React.Fragment>
    );
};

PromoPanelGroupDateSelector.propTypes = {
    isEditingDate: PropTypes.bool,
    defaultStartDate: PropTypes.any,
    defaultEndDate: PropTypes.any,
    scheduleChangeIndex: PropTypes.number,
    scheduleGroupIndex: PropTypes.string,
    handleCloseGroupDate: PropTypes.func,
    handleSaveGroupDate: PropTypes.func,
    fullPromoPanelUserTypeList: PropTypes.array,
    setConfirmationMessage: PropTypes.func,
    setIsConfirmOpen: PropTypes.func,
    setConfirmationMode: PropTypes.func,
    panelScheduleId: PropTypes.number,
    displayList: PropTypes.object,
};

PromoPanelGroupDateSelector.defaultProps = {
    previewGroup: [],
    previewContent: '',
    helpButtonLabel: 'Help',
    helpContent: 'test',
    scheduleChangeIndex: null,
    scheduleGroupIndex: null,
};

export default PromoPanelGroupDateSelector;
