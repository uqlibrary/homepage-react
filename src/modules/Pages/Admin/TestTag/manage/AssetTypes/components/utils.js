export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, rows: {}, row: {}, title: '' };

export const actionReducer = (_, action) => {
    switch (action.type) {
        case 'add':
            return {
                title: 'Add Asset Type',
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: { asset_type_id: 'auto' },
            };
        case 'edit':
            return { title: 'Edit Asset Type', isAdd: false, isEdit: true, isDelete: false, row: action.row };
        case 'clear':
            return { ...emptyActionState };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, row: action.row };
        default:
            throw `Unknown action '${action.type}'`;
    }
};
