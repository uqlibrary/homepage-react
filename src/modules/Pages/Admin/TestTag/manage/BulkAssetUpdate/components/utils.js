import { createLocationString } from '../../../helpers/helpers';

export const transformRow = row => {
    return row.map(line => {
        if (!!line?.asset_location) return line;
        return {
            ...line,
            asset_type_name: line?.asset_type?.asset_type_name ?? '',
            asset_location: !!line?.last_location
                ? createLocationString({
                      site: line.last_location.site_name,
                      building: line.last_location.building_name,
                      floor: line.last_location.floor_id_displayed,
                      room: line.last_location.room_id_displayed,
                  })
                : '',
        };
    });
};

export const transformRequest = formValues => {
    if (!!formValues.hasDiscardStatus) {
        formValues.hasLocation = false;
        formValues.hasAssetType = false;
    } else {
        formValues.hasDiscardStatus = false;
    }
    return {
        asset: formValues.asset_list.reduce((cumulative, current) => [...cumulative, current.asset_id], []),
        ...(!!formValues.hasLocation ? { asset_room_id_last_seen: formValues.location.room } : {}),
        ...(!!formValues.hasAssetType ? { asset_type_id: formValues.asset_type.asset_type_id } : {}),
        ...(!!formValues.hasDiscardStatus ? { is_discarding: 1, discard_reason: formValues.discard_reason } : {}),
    };
};

export const transformFilterRow = row => {
    return row.map(line => {
        if (!!line?.asset_id_displayed) return line;
        return {
            ...line,
            asset_id_displayed: line?.asset_barcode ?? '',
            asset_location: createLocationString({
                site: line.site_name,
                building: line.building_name,
                floor: line.floor_id_displayed,
                room: line.room_id_displayed,
            }),
        };
    });
};
