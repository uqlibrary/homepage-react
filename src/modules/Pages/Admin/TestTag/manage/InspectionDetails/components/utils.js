import moment from 'moment';
import locale from '../../../testTag.locale';
const dateFormat = locale.config.format.dateFormatNoTime;

export const transformRow = row => {
    return row.map(line => {
        if (!!row.user_name) return row;
        return {
            asset_id: line.asset_id,
            asset_id_displayed: line.asset_id_displayed,
            asset_type: line.asset_type?.asset_type_name,
            asset_status: line.asset_status,
            user_name: line.last_inspection?.user_name,
            inspect_date: !!line.last_inspection ? moment(line.last_inspection.inspect_date).format(dateFormat) : '',
            inspect_notes: line.last_inspection?.inspect_notes ?? '',
            inspect_fail_reason: line.last_inspection?.inspect_fail_reason ?? '',
            discard_reason: line.last_discard?.discard_reason ?? '',
            repairer_name: line.last_repair?.repairer_name ?? '',
            last_inspect_status: line.last_inspection?.inspect_status ?? '',
        };
    });
};

export const transformUpdateRequest = request => {
    const clonedRequest = structuredClone(request);
    const { asset_status: assetStatus, last_inspect_status: lastInspectStatus } = clonedRequest;
    const defaultValues = { inspect_notes: clonedRequest.inspect_notes };

    switch (assetStatus) {
        case locale.config.assetStatus.current:
            return defaultValues;
        case locale.config.assetStatus.failed:
            return { ...defaultValues, inspect_fail_reason: clonedRequest.inspect_fail_reason };
        case locale.config.assetStatus.outforrepair:
            return {
                ...defaultValues,
                inspect_fail_reason: clonedRequest.inspect_fail_reason,
                repairer_name: clonedRequest.repairer_name,
            };
        case locale.config.assetStatus.discarded:
            return {
                ...defaultValues,
                ...(lastInspectStatus === locale.config.inspectStatus.failed
                    ? { inspect_fail_reason: clonedRequest.inspect_fail_reason }
                    : {}),
                discard_reason: clonedRequest.discard_reason,
            };
        default:
            return request;
    }
};
