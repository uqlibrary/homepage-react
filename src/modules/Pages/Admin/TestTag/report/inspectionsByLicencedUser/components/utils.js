import locale from '../../../testTag.locale';
const moment = require('moment');

export const getNameStyles = (name, inspectorName, theme) => ({
    fontWeight:
        inspectorName.indexOf(name) === -1 ? theme.typography.fontWeightRegular : theme.typography.fontWeightMedium,
});

export const transformRow = row => {
    return row.map(line => ({
        ...line,
        start_date: !!line?.start_date ? moment(line.start_date).format(locale.config.format.dateFormatNoTime) : '--',
        end_date: !!line?.start_date ? moment(line.end_date).format(locale.config.format.dateFormatNoTime) : '--',
    }));
};
