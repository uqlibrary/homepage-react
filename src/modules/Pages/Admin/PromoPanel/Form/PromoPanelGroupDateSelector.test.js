import React from 'react';
import PromoPanelGroupDateSelector from './PromoPanelGroupDateSelector';
import { rtlRender, fireEvent } from '../../../../../../utils/test-utils';

const PromoList = [
    {
        usergroup_group: 'student',
        scheduled_panels: [
            {
                panel_schedule_id: 1,
                panel_schedule_start_time: '2022/01/01 00:00:00',
                panel_schedule_end_time: '2022/02/01 00:00:00',
            },
        ],
    },
];
const ConflictList = [
    {
        usergroup_group: 'student',
        scheduled_panels: [
            {
                panel_schedule_id: 1,
                panel_schedule_start_time: '2000/01/01 00:00:00',
                panel_schedule_end_time: '2080/02/01 00:00:00',
            },
        ],
    },
];
function setup(testProps = {}) {
    const props = {
        isEditingDate: true,
        defaultStartDate: '2000/01/01 00:00:00',
        defaultEndDate: '2000/01/02 23:59:00',
        handleCloseGroupDate: jest.fn(),
        handleSaveGroupDate: jest.fn(),
        setConfirmationMessage: jest.fn(),
        setIsConfirmOpen: jest.fn(),
        setConfirmationMode: jest.fn(),
        panelScheduleId: 0,
        fullPromoPanelUserTypeList: PromoList,
    };
    return rtlRender(<PromoPanelGroupDateSelector {...props} {...testProps} />);
}

describe('Promo Panel Group Date Selector', () => {
    it('Should render correctly', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });
    it('default values render correctly', () => {
        const { container } = setup();
        expect(container).toMatchSnapshot();
    });
    it('Should render date warnings', () => {
        const { getByTestId } = setup();
        expect(getByTestId('start-date-warning')).toBeInTheDocument();
        expect(getByTestId('end-date-warning')).toBeInTheDocument();
    });
    it('Should not render date warnings if dates are OK', () => {
        const { queryByTestId } = setup({
            defaultStartDate: '2040/01/01 00:00:00',
            defaultEndDate: '2040/01/02 23:59:00',
        });
        expect(queryByTestId('start-date-warning')).not.toBeInTheDocument();
        expect(queryByTestId('end-date-warning')).not.toBeInTheDocument();
    });

    it('Should handle close', () => {
        const handleCloseGroupDate = jest.fn();
        const { queryByTestId } = setup({
            handleCloseGroupDate: handleCloseGroupDate,
            defaultStartDate: '2040/01/01 00:00:00',
            defaultEndDate: '2040/01/02 23:59:00',
        });
        expect(queryByTestId('start-date-warning')).not.toBeInTheDocument();
        expect(queryByTestId('end-date-warning')).not.toBeInTheDocument();
        fireEvent.click(queryByTestId('admin-promopanel-group-button-cancel'));
        expect(handleCloseGroupDate).toBeCalled();
    });
    it('Should handle save', () => {
        const handleCloseGroupDate = jest.fn();
        const captureError = jest.fn();
        const { queryByTestId } = setup({
            handleCloseGroupDate: handleCloseGroupDate,
            setConfirmationMessage: captureError,
            defaultStartDate: '2040/01/01 00:00:00',
            defaultEndDate: '2040/01/02 23:59:00',
        });
        expect(queryByTestId('start-date-warning')).not.toBeInTheDocument();
        expect(queryByTestId('end-date-warning')).not.toBeInTheDocument();
        fireEvent.click(queryByTestId('admin-promopanel-group-button-save'));
        expect(captureError).not.toBeCalled();
    });
    it('Should capture conflict', () => {
        const handleCloseGroupDate = jest.fn();
        const captureError = jest.fn();
        const { queryByTestId } = setup({
            handleCloseGroupDate: handleCloseGroupDate,
            setConfirmationMessage: captureError,
            defaultStartDate: '2040/01/01 00:00:00',
            defaultEndDate: '2040/01/02 23:59:00',
            fullPromoPanelUserTypeList: ConflictList,
            scheduleChangeIndex: 0,
            scheduleGroupIndex: 'student',
        });
        expect(queryByTestId('start-date-warning')).not.toBeInTheDocument();
        expect(queryByTestId('end-date-warning')).not.toBeInTheDocument();
        fireEvent.click(queryByTestId('admin-promopanel-group-button-save'));
        expect(captureError).toBeCalled();
    });

    it('Cannot have start date afer end date', () => {
        const handleCloseGroupDate = jest.fn();
        const { queryByTestId, getByText } = setup({
            handleCloseGroupDate: handleCloseGroupDate,
            defaultStartDate: '2040/02/01 00:00:00',
            defaultEndDate: '2040/01/01 23:59:00',
            fullPromoPanelUserTypeList: ConflictList,
        });
        expect(queryByTestId('start-date-warning')).not.toBeInTheDocument();
        expect(queryByTestId('end-date-warning')).not.toBeInTheDocument();
        expect(getByText('Start Date cannot be after End Date.')).toBeInTheDocument();
    });
});
