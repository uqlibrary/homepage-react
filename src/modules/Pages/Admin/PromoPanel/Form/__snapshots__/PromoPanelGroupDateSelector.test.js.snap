// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo Panel Group Date Selector Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="lightboxTitle"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
          style="width: 500px;"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1qg91f3-MuiTypography-root-MuiDialogTitle-root"
            data-testid="panel-edit-date-title"
            id="lightboxTitle"
            style="position: relative; border-bottom: 1px solid #d7d1cc; font-size: 12px;"
          >
            <div
              class="MuiBox-root css-0"
              style="font-size: 1.5em; font-weight: bold; line-height: 1; margin: 0px;"
            >
              Edit Schedule Dates
            </div>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-uh0g7b-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1875xhu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="admin-promopanel-group-start-date"
                    id="admin-promopanel-group-start-date-label"
                  >
                    Start date
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-4cnau1-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      data-testid="admin-promopanel-group-start-date"
                      id="admin-promopanel-group-start-date"
                      inputmode="text"
                      placeholder="EEEE DD MMMM YYYY hh:mm aa"
                      type="text"
                      value="Sat 01 Jan 2000 12:00 am"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="Choose date, selected date is Jan 1, 2000"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="admin-promopanel-group-start-date-button"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CalendarIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="start-date-warning"
                  id="start-date-warning"
                >
                  This date is in the past.
                </div>
              </div>
              <div
                align="left"
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-uh0g7b-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1875xhu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="admin-promopanel-group-end-date"
                    id="admin-promopanel-group-end-date-label"
                  >
                    End date
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-4cnau1-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      data-testid="admin-promopanel-group-end-date"
                      id="admin-promopanel-group-end-date"
                      inputmode="text"
                      placeholder="EEEE DD MMMM YYYY hh:mm aa"
                      type="text"
                      value="Sun 02 Jan 2000 11:59 pm"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="Choose date, selected date is Jan 2, 2000"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="admin-promopanel-group-end-date-button"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CalendarIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="end-date-warning"
                  id="end-date-warning"
                >
                  This date is in the past.
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              />
              <div
                align="right"
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mhlsgc-MuiButtonBase-root-MuiButton-root"
                  data-testid="admin-promopanel-group-button-cancel"
                  style="margin-top: 10px;"
                  tabindex="0"
                  type="button"
                >
                  Cancel
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-t41e4y-MuiButtonBase-root-MuiButton-root"
                  data-testid="admin-promopanel-group-button-save"
                  style="margin-top: 10px;"
                  tabindex="0"
                  type="button"
                >
                  Ok
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Promo Panel Group Date Selector default values render correctly 1`] = `
<div
  aria-hidden="true"
/>
`;
